demos/mqtt/mqtt_demo_mutual_auth/mqtt_demo_mutual_auth.c

```diff
@@ -72,16 +83,34 @@
 #ifndef ROOT_CA_CERT_PATH
     #error "Please define path to Root CA certificate of the MQTT broker(ROOT_CA_CERT_PATH) in demo_config.h."
 #endif
-#ifndef CLIENT_CERT_PATH
-    #error "Please define path to client certificate(CLIENT_CERT_PATH) in demo_config.h."
-#endif
-#ifndef CLIENT_PRIVATE_KEY_PATH
-    #error "Please define path to client private key(CLIENT_PRIVATE_KEY_PATH) in demo_config.h."
-#endif
 #ifndef CLIENT_IDENTIFIER
     #error "Please define a unique CLIENT_IDENTIFIER."
 #endif
 
+/* AWS IoT message broker would require either a set of client certificate/private key
+ * or username/password to authenticate the client. */
+#ifndef CLIENT_USERNAME
+    #ifndef CLIENT_CERT_PATH
+        #error "Please define path to client certificate(CLIENT_CERT_PATH) in demo_config.h."
+    #endif
+    #ifndef CLIENT_PRIVATE_KEY_PATH
+        #error "Please define path to client private key(CLIENT_PRIVATE_KEY_PATH) in demo_config.h."
+    #endif
+#else
+
+/* If username is defined, a client password also would need to be defined for
+ * client authentication. */
+    #ifndef CLIENT_PASSWORD
+        #error "Please define client password(CLIENT_PASSWORD) in demo_config.h for client authentication based on username/password."
+    #endif
+
+/* AWS IoT MQTT broker port needs to be 443 for client authentication based on
```

Comment:
Can you fix the comment indentations for `CLIENT_PASSWORD` and `AWS_MQTT_PORT`?

Reply:
This is a problem with the formatting inside #ifdef and #endif blocks. I will see if there are any uncrustify.cfg updates that can be done to fix this issue.

Reply:
It seems like a known issue and fixed recently. https://github.com/uncrustify/uncrustify/issues/2756. I think it will be available in a future release. We would be able to just run the uncrustify after that update to fix the issue.